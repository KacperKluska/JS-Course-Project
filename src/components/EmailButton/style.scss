@import "../../colors";

.emailButton {
  width: 4rem;
  height: 4rem;

  font-size: 1.5rem;
  color: white;

  border: none;
  border-radius: 2rem;

  background-color: $blue-button-color;

  &:hover {
    background-color: $blue-button-color + #111;
  }
}

.emailForm {
  width: 400px;
  height: 500px;

  background-color: $background-color;

  border: 1px solid $primary-color;
}

.emailForm.active {
  animation: showEmailForm 0.5s ease-in-out forwards;
}

.navigation {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  width: 100%;

  padding: 0.5rem;

  color: white;

  background-color: $primary-color;

  label:hover {
    color: gray;
  }
}

.emailHelpInput {
  width: 90%;
  height: 2rem;

  padding: 0.5rem;
  margin: 1rem 1rem 0rem 1rem;

  outline: none;

  &:focus {
    border: 2px solid $blue-button-color;
  }
}

.emailHelpTextarea {
  min-height: 50%;

  resize: none;
}

.emailFormButton {
  width: 90%;

  outline: none;
  border: none;

  padding: 0.5rem;
  margin: 1.5rem 1rem 1rem 1rem;

  background-color: $blue-button-color;

  font-size: 1rem;
  color: white;

  &:hover {
    background-color: $blue-button-color + #111;
  }
}

.fixedToRight {
  position: fixed;
  bottom: 4rem;
  right: 2.5rem;
}

@media (max-width: 500px) {
  .fixedToRight {
    right: 1.5rem;
  }
}

@keyframes showEmailForm {
  0% {
    transform: translateX(100%);
  }
  100% {
    transform: translateX(0%);
  }
}
