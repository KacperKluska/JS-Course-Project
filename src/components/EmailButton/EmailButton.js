import React, { useState } from "react";
import "./style.scss";

const EmailButton = () => {
  const [buttonState, setButtonState] = useState(true);
  const [activeClass, setActiveClass] = useState("");

  const handleClick = () => {
    setButtonState((prev) => (prev = !prev));
    if (activeClass === "") {
      setActiveClass("active");
      setTimeout(() => {
        setActiveClass("");
      }, 500);
    }
  };

  return (
    <div className="fixedToRight">
      {!buttonState && (
        <form className={`emailForm ${activeClass}`}>
          <div className="navigation">
            Write email to us!
            <i className="icon-cancel" onClick={handleClick} />
          </div>
          <input className="emailHelpInput" placeholder="Full name" required />
          <input
            className="emailHelpInput"
            placeholder="Your email"
            type="email"
            required
          />
          <textarea
            className="emailHelpInput emailHelpTextarea"
            placeholder="Type here..."
            required
          />
          <button className="emailFormButton">Send</button>
        </form>
      )}
      {buttonState && (
        <button className="emailButton" onClick={handleClick}>
          <i className="icon-mail" />
        </button>
      )}
    </div>
  );
};

export default EmailButton;
