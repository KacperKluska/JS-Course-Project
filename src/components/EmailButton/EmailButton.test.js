import EmailButton from "./EmailButton";

let wrapper;

beforeEach(() => {
  wrapper = mount(<EmailButton />);
});

test("EmailButton snapshot with click simulated", () => {
  expect(wrapper).toMatchSnapshot();
  wrapper.find(".emailButton").simulate("click");
  expect(wrapper).toMatchSnapshot();
  wrapper.find(".icon-cancel").simulate("click");
  expect(wrapper).toMatchSnapshot();
});
