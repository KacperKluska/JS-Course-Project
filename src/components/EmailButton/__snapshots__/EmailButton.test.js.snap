// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailButton snapshot with click simulated 1`] = `
<div
  className="fixedToRight"
>
  <button
    className="emailButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="icon-mail"
    />
  </button>
</div>
`;

exports[`EmailButton snapshot with click simulated 2`] = `
<div
  className="fixedToRight"
>
  <form
    className="emailForm active"
  >
    <div
      className="navigation"
    >
      Write email to us!
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon-cancel"
        />
      </button>
    </div>
    <input
      className="emailHelpInput"
      placeholder="Full name"
      required={true}
    />
    <input
      className="emailHelpInput"
      placeholder="Your email"
      required={true}
      type="email"
    />
    <textarea
      className="emailHelpInput emailHelpTextarea"
      placeholder="Type here..."
      required={true}
    />
    <button
      className="emailFormButton"
      type="button"
    >
      Send
    </button>
  </form>
</div>
`;

exports[`EmailButton snapshot with click simulated 3`] = `
<div
  className="fixedToRight"
>
  <button
    className="emailButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="icon-mail"
    />
  </button>
</div>
`;
